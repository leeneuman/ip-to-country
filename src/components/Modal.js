import React, { useState } from 'react';
import Button from './button';
import SearchList from './searchList';

function Modal() {
	const [ itemsList, setItemsList ] = useState([]);

	const addRow = () => {
		setItemsList([ ...itemsList, itemsList.length + 1 ]);
	};

	const reset = () => {
		setItemsList([]);
	};

	return (
		<div>
			<button type="button" className="btn btn-primary" data-toggle="modal" data-target="#IPModal">
				Search IP's country
			</button>

			<div
				className="modal fade"
				id="IPModal"
				tabIndex="-1"
				role="dialog"
				aria-labelledby="ip to country modal"
				aria-hidden="true"
				data-backdrop="static"
			>
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<h5 className="modal-title" id="IPModal">
								IP Lookup
							</h5>
							<button
								type="button"
								className="close"
								data-dismiss="modal"
								aria-label="Close"
								onClick={reset}
							>
								<span aria-hidden="true">&times;</span>
							</button>
						</div>
						<div className="modal-body">
							<div>Enter one or more IP addresses and get their conutry</div>
							<Button label={'Add'} onClick={addRow} />
							<SearchList itemsList={itemsList} />
						</div>
					</div>
				</div>
			</div>
		</div>
	);
}

export default Modal;
