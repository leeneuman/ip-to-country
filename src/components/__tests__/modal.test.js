import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Modal from './../modal';

afterEach(cleanup);

test('should render modal component', () => {
	render(<Modal />);
	const itemElement = screen.getByTestId('modal');
	expect(itemElement).toBeInTheDocument();
});

test('matches snapshot', () => {
	const tree = renderer.create(<Modal />).toJSON();
	expect(tree).toMatchSnapshot();
});
