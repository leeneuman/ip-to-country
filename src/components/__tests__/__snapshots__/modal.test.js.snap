// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="modal"
>
  <button
    className="btn btn-primary"
    data-target="#IPModal"
    data-toggle="modal"
    type="button"
  >
    Search IP's country
  </button>
  <div
    aria-hidden="true"
    aria-labelledby="ip to country modal"
    className="modal fade"
    data-backdrop="static"
    id="IPModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="IPModal"
          >
            IP Lookup
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div>
            Enter one or more IP addresses and get their conutry
          </div>
          <button
            aria-label="Add"
            className="btn btn-primary mt-3"
            data-testid="button"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
              className="mr-1"
            >
              +
            </span>
            Add
          </button>
          <div
            className="mt-3 pt-3"
            data-testid="search-list"
            style={
              Object {
                "borderTop": "1px solid #dee2e6",
              }
            }
          >
            <div
              className="d-flex p-2 flex-column"
              data-testid="item-1"
            >
              <div
                className="d-flex align-items-center"
              >
                <div
                  className="d-flex align-items-center justify-content-center mr-3"
                  style={
                    Object {
                      "backgroundColor": "#e5e5e5",
                      "borderRadius": "50%",
                      "flexShrink": "0",
                      "height": "30px",
                      "width": "30px",
                    }
                  }
                >
                  1
                </div>
                <input
                  aria-label="ip address"
                  className="form-control mr-3 w-50"
                  data-testid="item-input-1"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
